app/controllers/sessions_controller.rb

```diff
@@ -1,4 +1,28 @@
 class SessionsController < ApplicationController
   def create
+    user_info = request.env['omniauth.auth']
+    name = user_info[:info][:name]
+    uid = user_info[:uid]
+    email = user_info[:info][:email]
+    token = user_info[:credentials][:token]
+
+    found_user = User.find_or_create_by(uid: uid) do |user|
+      user.uid = uid
+      user.email = email
+      user.token = token
+      user.name = name
+    end
+
+    if !User.exists?(uid: uid)
```

Comment:
Won't this always eval to false (and thus skip the post request to the back end) regardless of whether the user existed prior to the find_or_create_by on line 9? I think we'd have to either (a) do a find_or_initialize_by and instead of find_or_create_by, then check if the initialized user is a new record and either save & send a post request (if true) or just move on to setting the session uid (if false), or possibly (b) send the post request within the block for find_or_create_by, since the block will only be evaluated if it couldn't find the user and is therefore creating one.

Reply:
Good catch, I didn't realize we pushed up duplicate code!